pg_replicate/src/pipeline/batching/data_pipeline.rs

```diff
@@ -21,16 +21,14 @@ use super::BatchConfig;
 
 #[derive(Debug, Clone)]
 pub struct BatchDataPipelineHandle {
-    stream_stop: Arc<Notify>,
+    copy_tables_stream_stop: Arc<Notify>,
+    cdc_stream_stop: Arc<Notify>,
 }
 
 impl BatchDataPipelineHandle {
     pub fn stop(&self) {
-        // We want to notify all waiters that their streams have to be stopped.
-        //
-        // Technically, we should not need to notify multiple waiters since we can't have multiple
-        // streams active in parallel, but in this way we cover for the future.
-        self.stream_stop.notify_waiters();
```

Comment:
I found out that `notify_waiters` doesn't work when the stop is issued before the `notified` is called by at least one consumer. So, I used `notify_one` for each stream.

I could have also used a watch but it was not as ergonomic.